framework/db/QueryBuilder.php

```diff
@@ -1112,7 +1113,7 @@ public function buildInCondition($operator, $operands, &$params)
             return $this->buildSubqueryInCondition($operator, $column, $values, $params);
         }
 
-        $values = (array) $values;
+        $values = ArrayHelper::toArray($values);
```

Comment:
should set recursive parameter to false to avoid complete iteration over the values if it is already an array.